internal/cni/cni.go

```diff
@@ -195,44 +203,27 @@ func CmdAdd(args *skel.CmdArgs) error {
 	}
 
 	if cfg.Mode == "primary" {
-		deviceFile, err := tools.FilePathExists(constants.DeviceFile.Directory + constants.DeviceFile.Name)
+		ethInstalled, version, err := host.HasEthtool()
 		if err != nil {
-			logging.Errorf("cmdAdd(): Failed to locate deviceFile: %v", err)
+			logging.Warningf("cmdAdd(): failed to discover ethtool on host: %v", err)
 		}
-
-		if deviceFile {
-			deviceDetails, err = netHandler.GetDeviceFromFile(cfg.Device, constants.DeviceFile.Directory+constants.DeviceFile.Name)
-			if err != nil {
-				logging.Errorf("cmdAdd():- Failed to extract device map values: %v", err)
-				return err
-			}
-
-			ethInstalled, version, err := host.HasEthtool()
-			if err != nil {
-				logging.Warningf("cmdAdd(): failed to discover ethtool on host: %v", err)
-			}
-
-			if ethInstalled {
-				logging.Debugf("cmdAdd(): ethtool found on host")
-				logging.Debugf("\t" + version)
-				if deviceDetails != nil {
-					if deviceDetails.GetEthtoolFilters() != nil {
-						logging.Infof("cmdAdd(): applying ethtool filters on device: %s", cfg.Device)
-						ethtoolCommand := deviceDetails.GetEthtoolFilters()
-						iPAddr, err := extractIP(result)
-						if err != nil {
-							logging.Errorf("cmdAdd(): Error extracting IP from result interface %v", err)
-							return err
-						}
-						err = netHandler.SetEthtool(ethtoolCommand, cfg.Device, iPAddr)
-						if err != nil {
-							logging.Errorf("cmdAdd(): unable to executed ethtool filter: %v", err)
-							return err
-						}
-					} else {
-						logging.Debugf("cmdAdd(): ethtool filters have not been specified")
-					}
+		if ethInstalled {
```

Comment:
might be nicer here to first check if the NAD contains ethtool cmds and then check if ethtool is installed
i.e. kind of swap lines 210 and 213

If the user has ethtool cmds in the NAD and the host can't provide it, then we can log an error